taichi/runtime/gfx/runtime.cpp

```diff
@@ -560,6 +558,12 @@ void GfxRuntime::buffer_copy(DevicePtr dst, DevicePtr src, size_t size) {
   current_cmdlist_->buffer_copy(dst, src, size);
   submit_current_cmdlist_if_timeout();
 }
+void GfxRuntime::transition_image(DeviceAllocation image, ImageLayout layout) {
+  ImageLayout &last_layout = last_image_layouts_[image.alloc_id];
```

Comment:
I would use `.at` here to reduce the chance of error. `[]` operator may result in an insert with a silent initial value, which might not be what we want.

Reply:
It's a desired outcome actually. Images that have not been tracked will be initialized with undefined. Or should I make it more explicit with map.find?

Reply:
I'd prefer making it more explicit so that we make it easier for people down the line to maintain the code

Reply:
Also I would think we shouldn't do this in here. The GfxRuntime should track all the resources it use. If some resource not owned by the runtime is passed in this function, something might have gone wrong

Reply:
I see, I'm gonna delegate the allocation in `GfxRuntime`.